### ISSUE TITLE
Incorrect instance of GShowSingle for (f :.: g)

### ISSUE THREAD

**Issue by anka-213** [Lysxia/generic-data#16]
The left parameter (`f`) in `f :.: g` is not a generic representation, but a plain representation. Only the right parameter is a Generic representation. See example [here](https://hackage.haskell.org/package/base-4.12.0.0/docs/GHC-Generics.html#g:17).

The instance

```haskell
instance (GShowSingle Identity f, GShowSingle p g)
  => GShowSingle p (f :.: g) where
  ...
```
should instead be
```haskell
instance (Show1 f, GShowSingle p g)
  => GShowSingle p (f :.: g) where
  ...
```
similarly to the instance for `Rec1 f`.

----

A simple example of this would be trying to derive `Show1` for a type equivalent to `Compose`.

```haskell
newtype MyCompose f g a = MyCompose (f (g a))
  deriving Generic1
  deriving Show1 via Generically1 (MyCompose f g)
```
gives the error
```
    * Could not deduce (Generic.Data.Internal.Show.GShowSingle
                          Data.Functor.Identity.Identity f)
        arising from the 'deriving' clause of a data type 
declaration
      from the context: Show1 g
```

**Comment by Lysxia** [comment]
whoops. Thanks for catching this!

**Comment by Lysxia** [comment]
Fixed by 1bd6cfd4c379097d5554253fccc1b31a20fe85e7 Released in 0.6.0.1. Thanks again!
